### PR TITLE
Fix signature index issue

### PR DESCRIPTION
**Pull request by lachlansneff**
This fixed an emscripten regression.

### CODE REVIEW

File: lib/emscripten/src/lib.rs [hunk 1]
```diff
@@ -144,6 +144,8 @@ pub fn run_emscripten_instance(
         instance.call("___emscripten_environ_constructor", &[])?;
     }
 
+    println!("running emscripten instance");
```
Comment (bjfish) [inline lib/emscripten/src/lib.rs]: Remove println

File: lib/emscripten/tests/emtests/_common.rs [hunk 2]
```diff
@@ -30,14 +30,14 @@ macro_rules! assert_emscripten_output {
...
-         );
+        let output = capturer.end().unwrap().0;
+        let expected_output = include_str!($expected);
+
+        assert!(
+            output.contains(expected_output),
+            "Output: `{}` does not contain expected output: `{}`",
+            output,
+            expected_output
+        );
```
Comment (bjfish) [inline lib/emscripten/tests/emtests/_common.rs]: Is this formatted correctly?

File: lib/emscripten/src/lib.rs (hunk 1 above)
Comment (syrusakbary) [inline lib/emscripten/src/lib.rs]: Probably want a `debug!` here :)

File: lib/clif-backend/src/call/recovery.rs [hunk 3]
```diff
@@ -90,18 +90,9 @@ pub fn call_protected<T>(handler_data: &HandlerData, f: impl FnOnce() -> T) -> R
...
-                                    memory: MemoryIndex::new(0),
-                                    addr: addr as u32,
-                                }
-                            } else {
-                                // if there's an invalid access outside of the memory, including guard pages
-                                // just kill the process.
-                                panic!("invalid memory access, way out of bounds")
+                            RuntimeError::OutOfBoundsAccess {
+                                memory: MemoryIndex::new(0),
+                                addr: 0,
```
Comment (xmclark) [inline lib/clif-backend/src/call/recovery.rs]: this is slightly unrelated, but still good 👍

File: lib/runtime-core/src/instance.rs [hunk 4]
```diff
@@ -113,7 +113,7 @@ impl Instance {
                 .func_assoc
                 .get(*func_index)
                 .expect("broken invariant, incorrect func index");
-            let signature = SigRegistry.lookup_signature(sig_index);
+            let signature = &self.module.info.signatures[sig_index];
```
Comment (xmclark) [inline lib/runtime-core/src/instance.rs]: ❤️

**Pull request merged**